src/ff/field.rs

```diff
@@ -62,6 +53,7 @@ impl<F: Field> Serializable for F {
 pub trait BinaryField: Field + BooleanOps {}
 
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "enable-serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
my only concern (and maybe this is whatever) is that you've basically got the str repr of `FieldType` defined in 2 places. one in derived serde, and another in `AsRef`. so if you ever change one, you have to remember to change the other. and it would be hard to remember because it's not obvious that they're linked.

of course we do have tests that would check for this, but still.

Reply:
yea we probably don't need `AsRef` anywhere except in tests so I took the liberty to get rid of it